src/LibVLCSharp/Structures/MediaTrack.cs

```diff
@@ -284,7 +284,8 @@ internal MediaTrackList(IntPtr mediaTrackListPtr) : base(() => mediaTrackListPtr
         {
             get
             {
-                if (position > Count)
+                var count = Count;
+                if (count == 0 || position > count)
```

Comment:
Should likely be `if (position >= count)`. What if Count is 1 and `[1]` is called?

Reply:
> What if Count is 1 and `[1]` is called?

Hmm then that'd be an invalid call I guess? Count is 1-based while the indexer (`libvlc_media_tracklist_at`) is 0-based. Making me doubt now :D

Reply:
Added your suggestion just to be safe, though from my understanding that condition would never have ran considering the following list implementation
```
public bool MoveNext()
{
    position++;
    return position < (_mediaTrackList?.Count ?? 0);
}
```

Reply:
`MoveNext` is called when iterating in a `foreach` loop or when doing linq, but the indexer is not used in that case.

The proper test case would be to call `mediaTrackList[1]` directly, without checking the bounds.

Reply:
> MoveNext is called when iterating in a foreach loop or when doing linq, but the indexer is not used in that case.

Yes, it is (but it doesn't matter) :-) The `IEnumerator` states that after `MoveNext` is called when iterating over a list such as the media track list, `Current` is then called, which does `return _mediaTrackList[position]`.